java/src/test/java/ai/rapids/cudf/RmmTest.java

```diff
@@ -399,9 +409,28 @@ public void testPoolLimitNonPoolMode() {
         () -> Rmm.initialize(RmmAllocationMode.CUDA_DEFAULT, false, 1024, 2048));
   }
 
-  private static class AllocFailException extends RuntimeException {}
-  private static class AllocThresholdException extends RuntimeException {}
-  private static class DeallocThresholdException extends RuntimeException {}
+  @Test
+  public void testCudaAsyncMemoryResource() {
+    try {
+      Rmm.initialize(RmmAllocationMode.CUDA_ASYNC, false, 1024 * 1024L, 1024 * 1024L);
```

Comment:
It would be good to also test that the `CUDA_ASYNC` resource is not being setup with `CUDA_MANAGED_MEMORY`. This requires a new exception in `Rmm.initialize`

Reply:
Done.